### PR TITLE
Add a new way to manage secrets: secret templating

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Add a new way to manage secrets: secret templating
```

### CODE REVIEW

File: pkg/config/deployment.go [hunk 1]
```diff
@@ -40,6 +40,8 @@ type GenericDeploymentSpec struct {
 	// The maximum length of time to execute deployment before giving up.
 	// Default is 6h.
 	Timeout Duration `json:"timeout,omitempty" default:"6h"`
+	// List of encrypted secrets and targets that should be decoded before using.
+	Encryption *SecretEncryption `json:"encryption"`
```
Comment (nghialv) [inline pkg/config/deployment.go:44]: Tell me if you have a better field name for this.

Comment (nakabonne) [reply pkg/config/deployment.go:44]: I feel `encryption` is enough appropriate 👍

**Pull request merged**